src/tiled/mainwindow.cpp

```diff
@@ -1129,6 +1131,33 @@ void MainWindow::setFullScreen(bool fullScreen)
         setWindowState(windowState() & ~Qt::WindowFullScreen);
 }
 
+void MainWindow::toggleClearView(bool clearView)
+{
+    QList<QDockWidget*> docks = this->findChildren<QDockWidget*>();
+    QList<QToolBar*> toolbars = this->findChildren<QToolBar*>();
+
+    if (clearView) {
+        mHiddenDocks.clear();
+        mHiddenToolbars.clear();
+
+        for (auto dock : docks) {
+            if (dock->isVisible())
+                mHiddenDocks.push_back(dock);
+            dock->hide();
+        }
+        for (auto toolbar : toolbars) {
+            if (toolbar->isVisible())
+                mHiddenToolbars.push_back(toolbar);
```

Comment:
Please prefer to use the Qt-style members, `append` in this case, and not `push_back`, which is only there for STL compatibility.